examples/10_recipe.py

```diff
@@ -0,0 +1,451 @@
+processing steps) and its ranges of hyperparameters (the values that control
+the behavior of each step in the pipeline, such as the number of trees in a
+random forest).
+
+The |Recipe| brings 3 major benefits:
+
+- The data processing pipeline can be built step by step, while easily checking
+  the output of data transformations we have added so far. This makes
+  development faster and more interactive.
+
+- When we add a transformation, we can easily specify which columns it should
+  modify. Tabular data is heterogeneous and many processing steps only apply to
+  some of the columns in our dataframe.
+
+- When we add a processing step, we can specify a range for any of its
+  parameters, rather than a single value. Thus the ranges of hyperparameters to
+  consider for tuning are provided directly inside the estimator that uses
+  them. Moreover, these hyperparameters can be given human-readable names which
+  are useful for inspecting hyperparameter search results.
+
+.. admonition:: The Recipe has no ``fit()`` or ``predict()``
+
+    The recipe is not a scikit-learn estimator with ``fit`` and ``predict``
+    methods. Rather, it is a tool for configuring a scikit-learn estimator.
+    Once we have created a Recipe, we must call one of its methods such as
+    ``get_pipeline``, ``get_grid_search``, or ``get_randomized_search`` to
+    obtain the corresponding scikit-learn object.
+
+    For example:
+
+    - ``get_pipeline`` returns a scikit-learn |Pipeline| which applies all the
+      steps we have configured without any hyperparameter tuning.
+    - ``get_grid_search`` returns a scikit-learn |GridSearchCV| which uses the same
+      pipeline but runs a nested cross-validation loop to evaluate different
+      combinations of hyperparameters from the ranges we have specified and select
+      the best configuration.
+
+.. |Pipeline| replace::
+    :class:`~sklearn.pipeline.Pipeline`
+
+.. |GridSearchCV| replace::
+    :class:`~sklearn.model_selection.GridSearchCV`
+
+.. |RandomizedSearchCV| replace::
+    :class:`~sklearn.model_selection.GridSearchCV`
+
+.. |HGB| replace::
+    :class:`~sklearn.ensemble.HistGradientBoostingRegressor`
+
+.. |TargetEncoder| replace::
+    :class:`~sklearn.preprocessing.TargetEncoder`
+
+.. |Recipe| replace::
+     :class:`~skrub.Recipe`
+
+.. |TableReport| replace::
+     :class:`~skrub.TableReport`
+
+.. |DatetimeEncoder| replace::
+     :class:`~skrub.DatetimeEncoder`
+
+.. |MinHashEncoder| replace::
+     :class:`~skrub.MinHashEncoder`
+
+"""
+
+
+# %%
+# Getting a preview of the transformed data
+# -----------------------------------------
+#
+# The |Recipe| can be initialized with the full dataset (including the
+# prediction target, ``y``). We can tell it which columns constitute the target
+# and should be kept separate from the rest.
+
+# %%
+from skrub import Recipe, datasets
+
+dataset = datasets.fetch_employee_salaries()
```

Comment:
I looked a bit but haven't found a good replacement yet. but as I suspect we will merge the fraud data example before this one, maybe employee salaries can be replaced with that one

Reply:
Ok, to do so we need to take care of the join operation with the recipe first, right?

Reply:
good point, so that would come later. if anyone has suggestions for another dataset for this example I'd be happy to diversify a bit from employee salaries.

also, in employee salaries should we remove the "year first hired" in the fetcher? both here and in the tablevectorizer examples, the datetime encoder isn't useful because the feature it extracts has already been inserted in the dataset